File: solutions/python/bob/1/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if hey_bob.endswith('?') and not hey_bob.isupper():
        return "Sure."
    elif hey_bob.isupper() and not hey_bob.endswith('?'):
        return "Whoa, chill out!"
    elif hey_bob.endswith('?') and hey_bob.isupper():
        return "Calm down, I know what I'm doing!"
    elif not hey_bob:
        return "Fine. Be that way!"
    else:
        return "Whatever."

File: solutions/javascript/bob/1/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export function hey(message) {
  const msg = message.trim();

  // Silence
  if (msg === '') {
    return 'Fine. Be that way!';
  }

  const isQuestion = msg.endsWith('?');

  const hasLetters = /[a-zA-Z]/.test(msg);
  const isYelling = hasLetters && msg === msg.toUpperCase();

  // Yelling question
  if (isYelling && isQuestion) {
    return "Calm down, I know what I'm doing!";
  }

  // Yelling
  if (isYelling) {
    return 'Whoa, chill out!';
  }

  // Question
  if (isQuestion) {
    return 'Sure.';
  }

  // Anything else
  return 'Whatever.';
}
